test/pytest_control.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+error_pytest_control() {
+    echo -e "ERROR: $*. Aborting." >&2
+    return 1
+}
+
+setup_pytest_venv() {
+    python3 -m venv test/venv
+    if [[ -f "test/venv/bin/activate" ]]; then
+        echo "Starting the virtual environment now ..."
+        # shellcheck disable=SC1091
+        source test/venv/bin/activate
+        pip install pytest six
+    else
+        error_pytest_control "The directory '${PWD}/test/venv' does not exist. Cannot enter the virtual environment"
+    fi
+}
+
+run_pytest() {
+    if [[ -f "test/venv/bin/activate" ]]; then
```

Comment:
should you also check if the virtualenv is already active?

Reply:
I added the check, but it shouldn't really matter. By executing the script it's performed in a subprocess, which won't have the virtualenv active even if it is active on the parent process.